function/expression.go

```diff
@@ -65,15 +65,26 @@ func (f MetricFunction) Evaluate(context EvaluationContext,
 // fetchCounter is used to count the number of fetches remaining in a thread-safe manner.
 type FetchCounter struct {
 	count *int32
+	limit int
 }
 
 func NewFetchCounter(n int) FetchCounter {
 	n32 := int32(n)
 	return FetchCounter{
 		count: &n32,
+		limit: n,
 	}
 }
 
+// limit returns the max # of fetches allowed by this counter.
+func (c FetchCounter) Limit() int {
+	return c.limit
+}
+
+func (c FetchCounter) Current() int {
```

Comment:
i fixed it. it was supposed to return the current # of fetches. added tests for it.